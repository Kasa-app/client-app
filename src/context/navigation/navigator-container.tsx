import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LoginPage from '../../screens/authentification/login/login-page-view';
import SignUpPage from '../../screens/authentification/signup/signup-page';
import StartupPage from '../../screens/authentification/startup/startup-page';
import ServiceDetailsPage from '../../screens/service-details/service-details-page';
import { TabsNavigation } from '../../components/Bottom-Navigation-View/Tabs-navigation/Bottom/tabs-navigation-view';
import ServiceList from '../../screens/service-list/service-list-page';
import SearchJobbersPage from '../../screens/search-jobbers/search-jobbers';
import JobberProfilePage from '../../screens/jobber-profile/jobber-profile';
import BookingPage from '../../screens/booking/booking-page';
import LaunchPage from '../../screens/launch-page/launch-page';
import Chat from '../../components/Bottom-Navigation-View/Chat/Chat-View/chat';

const Stack = createNativeStackNavigator();

export const NavigatorContainer = () => {
    return (
        <NavigationContainer>
            <Stack.Navigator
                screenOptions={{
                    headerShown: false,
                    presentation: 'modal',
                    animationTypeForReplace: 'push',
                    animation: 'slide_from_bottom',
                    animationDuration: 200
                }}
                initialRouteName='LaunchScreen'
            >
                <Stack.Screen name="LaunchScreen" component={LaunchPage} />
                <Stack.Screen name="StartupScreen" component={StartupPage} />
                <Stack.Screen name="LoginScreen" component={LoginPage} />
                <Stack.Screen name="SignUpScreen" component={SignUpPage} />
                <Stack.Screen name="ServiceDetailsScreen" component={ServiceDetailsPage} />
                <Stack.Screen name="BookingScreen" component={BookingPage} />
                <Stack.Screen name="ChatScreen" component={Chat} />
                <Stack.Screen name="TabsNavigationScreen" component={TabsNavigation} />
                <Stack.Screen name="ServiceListScreen" component={ServiceList} />
                <Stack.Screen name="SearchJobbersScreen" component={SearchJobbersPage} />
                <Stack.Screen name="JobberProfileScreen" component={JobberProfilePage} />
            </Stack.Navigator>
        </NavigationContainer>
    );
};
