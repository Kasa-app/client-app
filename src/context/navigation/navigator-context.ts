import { createContext } from "react";
import { NavigatorContextType } from "../../models/contextType/NavigatorContextType";

export const NavigatorContext = createContext<NavigatorContextType>({
  activeStep: null,
  setActiveStep: () => { },
});