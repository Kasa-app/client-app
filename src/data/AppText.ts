export const APP_TEXT = {
    authentification: {
        startup: {
            welcome_text: "Bienvenue sur Kasa",
            pickupLine1: "Gagnez du temps, √©conomisez de l'√©nergie",
            pickupLine2: "Commencez d√®s aujourd'hui !",
            inscription_text: "S'inscrire",
            connexion_text: "Se connecter",
        },
        signUp: {
            create_account_text: 'Cr√©ez un compte',
            ask_email_text: 'Quel est votre courriel ?',
            ask_user_name_text: 'Quel est votre nom et pr√©nom ?',
            ask_user_birthdate: 'Quel est votre date de naissance ?',
            ask_user_password: 'Quel mot de passe utiliserez-vous sur Kasa ?',
            email_signUp_text: 'S\'inscrire avec un email',
            social_media_signUp_text: 'ou inscrivez-vous avec',
            already_have_account_text: 'Appuyez ici si vous avez d√©ja un compte !',
            continue_text: 'Continuer',
            create_account_button_text:'Cr√©er mon compte',
            checkBox_text: 'J\'ai lu et j\'accepte les conditions g√©n√©rales d\'utilisation \n ainsi que la politique de protection des donn√©es personnelles',
            email_place_holder_text: 'johndoe@gmail.com',
            lastName_place_holder_text: 'Nom',
            firstName_place_holder_text: 'Pr√©nom',
            birthdate_place_holder_text: '2023-01-01',
            birthdate_restriction_text: 'Vous devez avoir au moins 18 ans pour acc√©der √† Kasa',
            password_place_holder_text: 'Mot de passe',
            password_not_match_error:'Les mots de passes ne correspondent pas',
            password_spec_error: 'Le mot de passe ne respecte pas les crit√®res ci-dessous',
            email_error: 'Adresse courriel invalide',
            email_already_exist:'Votre adresse courriel existe d√©ja',
            validate_email_error: 'Une erreur est survenue lors de la validation de votre adresse courriel',
            name_input_error:'Le champ doit avoir entre 3 et 20 caract√®res et ne doit contenir aucun caract√®re sp√©cial',
            confirmPassword_place_holder_text: 'Confirmer le mot de passe',
            password_restriction_text:
                `Le mot de passe doit contenir au moins:
                
- Une lettre majuscule. ex: B
- Une lettre minuscule. ex: c
- Un chiffre. ex: 1
- Le mot de passe doit contenir au moins 8 caract√®res.
            
exemple de mot de passe: Azerty123456
            `
      ,

    },
    signIn: {
      question_forget_password: 'Mot de passe oubli√© ?',
      social_media_signIn_text: 'ou connectez-vous avec',
      not_have_account_text: "Appuyez ici si vous n'avez pas de compte !",
      message_bienvenue: "Bienvenue",
      adresse_email: "Adresse courriel",
      mot_de_passe: "Mot de passe",
      connexion: "Connexion",
      login_failed: "La connexion a √©chou√©. Veuillez v√©rifier vos informations de connexion et r√©essayer.",
    },
  },
  booking: {
    address: 'Adresse',
    back_to_home: 'Retourner √† l\'accueil',
    booking_error: 'Une erreur est survenue lors de la r√©servation',
    postal_code: 'Code postal',
    continue: 'Continuer',
    comments: 'Commentaires',
    no_comment: "Il n'y a aucun commentaire pour le moment",
    search_jobber: 'Rechercher des jobbers',
    services_done: 'A r√©alis√© plus de',
    phone_number_placeholder: '819-XXX-XXXX',
    services: 'services',
    see_profile: 'Voir le profil',
    member_since: 'Membre depuis',
    no_jobbers: 'Malheureusement, il n\'y a aucun jobber disponible pour le moment. Veuillez r√©essayer plus tard.',
    languages: 'Langues parl√©es:',
    to_book: 'R√©servez',
    skills: 'Habilet√©s et exp√©riences',
    booking_success: 'Votre r√©servation a √©t√© effectu√©e avec succ√®s!',
    steps: {
        photos: 'Souhaitez-vous ajouter des photos ? (optionnel)',
        estimated_duration: 'Estimation de la dur√©e de service',
        address: 'Adresse de la prestation',
        pickupAddress: "Quel est l'adresse de d√©part ?",
        date: 'Quel jour vous convient le mieux ?',
        drop_address: "Quel est l'adresse d'arriv√©e ?",
        come_with_truck: "Le jobber doit-il venir avec son camion ?",
        own_vehicle: "Le jobber doit-il poss√©der un v√©hicule ?",
        own_mower: "Le jobber doit-il apporter sa propre tondeuse ?",
        frequency: "Fr√©quence de la demande ?",
        keep_duration: "Quand commence et se termine la garde ?",
        phone_number: "√Ä quel num√©ro de t√©l√©phone le jobber pourra-t-il vous joindre ?",
        more_details: "D√©tails suppl√©mentaires (optionnel)",
        jobber_counts: "Combien de prestataires vous faut-il ?",
        time: "Quelle heure vous convient le mieux ?",
        warning_time_min:"L'heure de d√©but doit √™tre apr√®s ou √† 6h du matin",
        warning_time_max:"L'heure de fin doit √™tre avant ou √† 21h00",
        package_delivery: "√Ä quel adresse le colis doit-il √™tre livr√© ?",
        more_than_one_day: "Envisagez-vous une dur√©e de service sup√©rieure √† une journ√©e ?",
        date_debut: "√Ä quel date souhaitez-vous que le service d√©bute ?",
        date_fin: "√Ä quel date souhaitez-vous que le service prenne fin ?"
    },

    stepKey:{
      jobber_counts: "jobberCounts",
      estimated_duration: "estimatedDuration",
      more_than_one_day:"moreThanOneDay"
    }
	},
  components: {
    show_more: 'Voir plus',
    show_less: 'Voir moins',
    retry: 'R√©essayer',
  },
  errors: {
    service_details: 'Une erreur innatendue est survenue lors du chargement des d√©tails du service.',
  },
  profile: {
    profile: 'Profil',
    manage_account: 'G√©rer mon compte',
    notifications: 'Notifications',
    options: 'Options',
    member_since: 'Membre depuis',
    help: 'Aide et assistance',
    logout: 'Se d√©connecter'
  },
  maintenance: {
    title: 'Sous d√©veloppement',
    message: "Oops! üõ†Ô∏è Ce service est en cours de d√©veloppement et n'est pas disponible pour le moment. Nous travaillons fort pour le rendre disponible sous peu. Revenez bient√¥t !",
  },
  services: {
    estimation_taux_horaire: 'Estimation du taux horaire:',
    estimation_duree_service: 'Estimation de la dur√©e du service:',
    said: 'a dit:',
    start_booking: 'Commencer ma r√©servation',
  },
  homePage: {
    categorie_section: 'Nos univers de categories',
    popular_services_section: 'Nos services populaires',
    recommended_services_section: 'Services recommand√©s',
    search_placeHolder: 'Rechercher un service'
  },
  history:{
    reservations_title:'Mes r√©servations',
    service_name: 'Nom du service',
    date_hour_title:'Date et heure',
    adresse_title:'Adresse',
    button_title:'G√©rer ma demande'
  }
}