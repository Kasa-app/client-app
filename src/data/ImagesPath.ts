const appLogo = require('../assets/app-logo.png');
const loginImage = require('../assets/login-image.jpg');
const headerLogo = require('../assets/header-logo.png');
const backButton = require('../assets/back-button.png');
const appleLogo = require('../assets/apple-logo.png');
const facebookLogo = require('../assets/facebook-logo.png');
const gmailLogo = require('../assets/gmail-logo.png');
const userAvatar = require('../assets/user-avatar.png');
const homeLogo = require('../assets/home-logo.png');
const chatLogo = require('../assets/chat-logo.png');
const historyLogo = require('../assets/history-logo.png');
const calendarLogo = require('../assets/calendar-logo.png');
const accountLogo = require('../assets/account-logo.png');
const findServiceLogo = require('../assets/find-service-logo.png');
const notificationLogo = require('../assets/notification-logo.png');
const homePic = require('../assets/home-pic.jpg')
const searchIcon = require('../assets/search-icon.png')
const nextArrow = require('../assets/next-arrow.png')
const sendButton = require('../assets/send-button.png')
const userImgTest = require('../assets/user-img-test.png')
export {
    appLogo,
    searchIcon,
    nextArrow,
    findServiceLogo,
    homePic,
    historyLogo,
    notificationLogo,
    calendarLogo,
    accountLogo,
    chatLogo,
    homeLogo,
    loginImage,
    headerLogo,
    backButton,
    appleLogo,
    facebookLogo,
    gmailLogo,
    userAvatar,
    userImgTest,
    sendButton
};