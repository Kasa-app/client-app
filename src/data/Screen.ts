export const Screen = {
  authentification: {
    login: 'LoginScreen',
    startup: 'StartupScreen',
    signUp: 'SignUpScreen',
  },
  booking: 'BookingScreen',
  searchJobbers: 'SearchJobbersScreen',
  jobberProfile: 'JobberProfileScreen',
  serviceDetails: 'ServiceDetailsScreen',
  tabsNavigation:'TabsNavigationScreen',
  serviceList:'ServiceListScreen',
  chat:'ChatScreen'
};