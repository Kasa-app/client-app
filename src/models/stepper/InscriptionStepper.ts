export enum InscriptionStepper {
  SignUpWithEmail = 'signUpWithEmail',
  UserEmail = 'User Email',
  UserName = 'User Name',
  UserBirthdate = 'User birthdate',
  UserPassword = 'User password',
}