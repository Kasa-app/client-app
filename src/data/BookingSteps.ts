import { APP_TEXT } from "./AppText";
import TextArea from "../components/Booking/MoreDetails/MoreDetails";
import Empty from "../components/Empty/Empty";
import TimeInput from "../components/Booking/TimeInput/time-input";
import Input from "../components/Booking/PhoneNumber/phoneNumber";
import DateTimePicker from "../components/Booking/DateTimePicker/date-time-picker";
import { isCanadianPostalCodeValid, isPhoneNumberValid } from "../utils/BookingUtils";
import { BookingStepType } from "../models/booking/BookingStepType";
import AddressInput from "../components/Booking/AddressInput/address-input";
import PhoneNumber from "../components/Booking/PhoneNumber/phoneNumber";

const BOOKING_STEPS: Record<string, BookingStepType> = {
  photos: {
    stepKey: 'photos',
    stepName: APP_TEXT.booking.steps.photos,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  estimated_duration: {
    stepKey: 'estimatedDuration',
    stepName: APP_TEXT.booking.steps.estimated_duration,
    component: TimeInput,
    inputProps: { isEstimatedService: true },
    onValidate: undefined,
  },
  address: {
    stepKey: 'address',
    stepName: APP_TEXT.booking.steps.address,
    component: AddressInput,
    inputProps: { postalCodeKey: 'postalCode' },
    onValidate: undefined,
  },
  date: {
    stepKey: 'date',
    stepName: APP_TEXT.booking.steps.date,
    component: DateTimePicker,
    inputProps: {},
    onValidate: undefined,
  },
  pickup_address: {
    stepKey: 'pickupAddress',
    stepName: APP_TEXT.booking.steps.pickup_address,
    component: Input,
    inputProps: {},
    onValidate: undefined,
  },
  come_with_truck: {
    stepKey: 'comeWithTruck',
    stepName: APP_TEXT.booking.steps.come_with_truck,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  own_vehicle: {
    stepKey: 'ownVehicle',
    stepName: APP_TEXT.booking.steps.own_vehicle,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  own_mower: {
    stepKey: 'ownMower',
    stepName: APP_TEXT.booking.steps.own_mower,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  frequency: {
    stepKey: 'frequency',
    stepName: APP_TEXT.booking.steps.frequency,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  keep_duration: {
    stepKey: 'keepDuration',
    stepName: APP_TEXT.booking.steps.keep_duration,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  phone_number: {
    stepKey: 'phoneNumber',
    stepName: APP_TEXT.booking.steps.phone_number,
    component: PhoneNumber,
    inputProps: { placeholder: APP_TEXT.booking.phone_number_placeholder },
    onValidate: undefined,
  },
  more_details: {
    stepKey: 'moreDetails',
    stepName: APP_TEXT.booking.steps.more_details,
    component: TextArea,
    inputProps: {},
    onValidate: undefined,
  },
  jobber_counts: {
    stepKey: 'jobberCounts',
    stepName: APP_TEXT.booking.steps.jobber_counts,
    component: Empty,
    inputProps: {},
    onValidate: undefined,
  },
  time: {
    stepKey: 'time',
    stepName: APP_TEXT.booking.steps.time,
    component: TimeInput,
    inputProps: {},
    onValidate: undefined,
  },
  package_delivery: {
    stepKey: 'packageDelivery',
    stepName: APP_TEXT.booking.steps.package_delivery,
    component: Input,
    inputProps: {},
    onValidate: undefined,
  },
};

const GENERIC_STEPS = [
  BOOKING_STEPS.date,
  BOOKING_STEPS.time,
  BOOKING_STEPS.address,
  BOOKING_STEPS.phone_number,
  BOOKING_STEPS.more_details
];

export const BOOKING_SERVICE_STEPS = {
  demenagement_residentiel: {
    steps: [...GENERIC_STEPS]
  },
  demenagement_et_ameublement: {
    steps: [...GENERIC_STEPS]
  },
  aide_a_un_demenagement: {
    steps: [...GENERIC_STEPS]
  },
  garde_de_chien: {
    steps: [...GENERIC_STEPS]
  },
  garde_de_chat: {
    steps: [...GENERIC_STEPS]
  },
  promenade_de_chien: {
    steps: [...GENERIC_STEPS]
  },
  toilettage_pour_chien: {
    steps: [...GENERIC_STEPS]
  },
  renovation_de_salle_de_bain: {
    steps: [...GENERIC_STEPS]
  },
  renovation_de_sous_sol: {
    steps: [...GENERIC_STEPS]
  },
  renovation_de_linterieur: {
    steps: [...GENERIC_STEPS]
  },
  renovation_de_lexterieur: {
    steps: [...GENERIC_STEPS]
  },
  peinture_interieure: {
    steps: [...GENERIC_STEPS]
  },
  decoration_interieure: {
    steps: [...GENERIC_STEPS]
  },
  transport_vers_laeroport: {
    steps: [...GENERIC_STEPS]
  },
  courses: {
    steps: [...GENERIC_STEPS]
  },
  livrer_un_colis: {
    steps: [...GENERIC_STEPS]
  },
  cuisinier_personnel: {
    steps: [...GENERIC_STEPS]
  },
  cours_de_cuisine_a_domicile: {
    steps: [...GENERIC_STEPS]
  },
  livraison_de_repas_faits_maison: {
    steps: [...GENERIC_STEPS]
  },
  coiffeur_coiffeuse: {
    steps: [...GENERIC_STEPS]
  },
  maquilleur_maquilleuse: {
    steps: [...GENERIC_STEPS]
  },
  reparation_de_plombeire: {
    steps: [...GENERIC_STEPS]
  },
  reparation_de_toiture: {
    steps: [...GENERIC_STEPS]
  },
  reparation_de_chauffage_et_de_climatisation: {
    steps: [...GENERIC_STEPS]
  },
  reparation_delectromenagers: {
    steps: [...GENERIC_STEPS]
  },
  machine_a_laver: {
    steps: [...GENERIC_STEPS]
  },
  lave_vaisselle: {
    steps: [...GENERIC_STEPS]
  },
  refrigerateur: {
    steps: [...GENERIC_STEPS]
  },
  four: {
    steps: [...GENERIC_STEPS]
  },
  pose_dune_television: {
    steps: [...GENERIC_STEPS]
  },
  serrure: {
    steps: [...GENERIC_STEPS]
  },
  abris_de_jardin: {
    steps: [...GENERIC_STEPS]
  },
  trampoline: {
    steps: [...GENERIC_STEPS]
  },
  montage_de_meubles: {
    steps: [...GENERIC_STEPS]
  },
  montage_de_lit: {
    steps: [...GENERIC_STEPS]
  },
  montage_de_canape: {
    steps: [...GENERIC_STEPS]
  },
  menage_a_domicile: {
    steps: [...GENERIC_STEPS]
  },
  nettoyage_de_vitres: {
    steps: [...GENERIC_STEPS]
  },
  nettoyage_de_tapis: {
    steps: [...GENERIC_STEPS]
  },
  tondre_la_pelouse: {
    steps: [...GENERIC_STEPS]
  },
  pose_de_gazon: {
    steps: [...GENERIC_STEPS]
  },
  ramassage_de_feuilles: {
    steps: [...GENERIC_STEPS]
  },
  deneiger: {
    steps: [...GENERIC_STEPS]
  },
  entretien_de_piscine_exterieur: {
    steps: [...GENERIC_STEPS]
  },
  coaching_sportif_personnel: {
    steps: [...GENERIC_STEPS]
  },
  coaching_en_nutrition: {
    steps: [...GENERIC_STEPS]
  },
  coaching_en_yoga: {
    steps: [...GENERIC_STEPS]
  },
};
