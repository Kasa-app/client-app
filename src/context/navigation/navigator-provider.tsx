import React, { useState } from 'react';
import { NavigatorContext } from './navigator-context';
import { InscriptionStepper } from '../../models/stepper/InscriptionStepper';
import { NavigatorContextType } from '../../models/contextType/NavigatorContextType';

export const NavigatorProvider = ({ children }: any) => {
  const [activeStep, setActiveStep] = useState(InscriptionStepper.SignUpWithEmail);
  const context: NavigatorContextType =
  {
    activeStep,
    setActiveStep,
  };

  return (
    <NavigatorContext.Provider value={context}>
      {children}
    </NavigatorContext.Provider>
  );
};

