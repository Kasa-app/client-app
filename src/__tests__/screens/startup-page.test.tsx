import React from 'react';
import { it } from '@jest/globals';
import { act, render } from '@testing-library/react-native';
import { NavigatorContext } from '../../context/navigation/navigator-context';
import { Screen } from '../../data/Screen';
import { TouchableOpacity } from 'react-native';
import { InscriptionStepper } from '../../models/stepper/InscriptionStepper';
import { create } from 'react-test-renderer';
import StartupPage from '../../screens/authentification/startup/startup-page';

const mockNavigatorContextValue = {
    routeName: '',
    setRouteName: jest.fn(),
    activeStep: InscriptionStepper.UserName,
    setActiveStep: jest.fn(),
  };
  const MockNavigatorContextProvider = ({ children }: any) => (
    <NavigatorContext.Provider value={mockNavigatorContextValue}>
        {children}
    </NavigatorContext.Provider>
  );

describe('StartupPage', () => {
    const tree = create(<StartupPage />);

    it('snapshot', () => {
        expect(tree).toMatchSnapshot();
    });

    it('should set signUp routeName when the "Inscription" button is pressed', async () => {
        const mockNavigateToScreen = jest.fn();
        const component = render(<StartupPage navigation={{ navigate: mockNavigateToScreen }} />,);
        const buttons = component.root.findAllByType(TouchableOpacity);
        const continueBtn = buttons[0].props;
        await act(async () => await continueBtn.onPress());

        expect(mockNavigateToScreen).toHaveBeenCalledWith(Screen.authentification.signUp);
    });
});
