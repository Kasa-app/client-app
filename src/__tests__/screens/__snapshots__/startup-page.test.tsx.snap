// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartupPage snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#C27129",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <Image
          source={
            {
              "testUri": "../../../src/assets/login-image.jpg",
            }
          }
          style={
            {
              "height": 1114,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/app-logo.png",
            }
          }
          style={
            {
              "height": 66,
              "marginStart": "5%",
              "marginTop": "4%",
              "width": 70,
            }
          }
        />
        <Text
          style={
            {
              "color": "#FFF",
              "fontFamily": "AvenirNext-Bold",
              "fontSize": 27,
              "marginEnd": "10%",
              "marginTop": "7%",
            }
          }
        >
          Bienvenue sur Kasa
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginStart": "10%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontFamily": "AvenirNext-Regular",
              "fontSize": 15,
              "marginEnd": "10%",
              "marginTop": "4%",
            }
          }
        >
          Gagnez du temps, économisez de l'énergie
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontFamily": "AvenirNext-Regular",
              "fontSize": 15,
              "marginEnd": "10%",
            }
          }
        >
          Commencez dès aujourd'hui !
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginStart": "14%",
            "marginTop": "6%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E67918",
              "borderRadius": 4,
              "height": "100%",
              "opacity": 1,
              "paddingBottom": 5,
              "paddingTop": 5,
              "width": "35%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFF",
                "fontFamily": "AvenirNext-Bold",
                "fontSize": 14,
              }
            }
          >
            S'inscrire
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#E67918",
              "borderRadius": 4,
              "borderWidth": 2,
              "height": "100%",
              "marginStart": "15%",
              "opacity": 1,
              "paddingBottom": 5,
              "paddingTop": 5,
              "width": "35%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFF",
                "fontFamily": "AvenirNext-Bold",
                "fontSize": 14,
              }
            }
          >
            Se connecter
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "height": 40,
          }
        }
      />
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
